backend/src/intervals/VaultSnapshot.ts

```diff
@@ -73,5 +158,114 @@ ponder.on('VaultSnapshot:block', async ({ event, context }) => {
         vault: vault.contractAddress,
       })
       .onConflictDoUpdate(snapshot)
+
+    // vault apy computation
+    try {
+      // Fetch snapshots for this vault and yield pool, ordered by timestamp
+      const historicalSnapshots = await context.db.sql
+        .select()
+        .from(vaultSnapshot)
+        .where(
+          and(
+            eq(vaultSnapshot.vault, vault.contractAddress),
+            eq(vaultSnapshot.chainId, vault.chainId),
+            eq(vaultSnapshot.yieldPool, vault.yieldPool)
+          )
+        )
+        .orderBy(vaultSnapshot.timestamp)
+        .execute()
```

Comment:
I would move that _inside_ `selectSnapshot` so that you don't select data that's not needed.
the problem with the current approach is that this `select` will return an increasingly large amount of data and that will eventually break.

You would need to add a a new `AND` conditions to return only snapshots that are greater than the `intervalSeconds`.